src/inmanta/env.py

```diff
@@ -696,11 +696,24 @@ def init_env(self) -> None:
                 raise VenvCreationFailedError(msg=f"Unable to create new virtualenv at {self.env_path}")
             LOGGER.debug("Created a new virtualenv at %s", self.env_path)
         elif not os.path.exists(self._path_pth_file):
-            if not os.path.exists(self.site_packages_dir):
-                # the venv hosts a different python version than the running process
-                raise VenvActivationFailedError(
-                    msg=f"Unable to use virtualenv at {self.env_path} due to python version mismatch"
-                )
+            # make sure the venv hosts the same python version as the running process
+            if sys.platform != "win32":
+                # on linux based systems, the python version is in the name of the executable:
+                if not os.path.exists(self.site_packages_dir):
+                    raise VenvActivationFailedError(
+                        msg=f"Unable to use virtualenv at {self.env_path} due to python version mismatch"
+                    )
+            else:
+                python_subprocess = subprocess.Popen([self.python_path, "--version"], stdout=subprocess.PIPE, encoding="utf-8")
```

Comment:
Any reason to not use `subprocess.check_output()`? It would make this easier to read imo.